librad/src/net/peer.rs

```diff
@@ -503,13 +517,18 @@ where
 {
     type Update = Gossip;
 
-    async fn put(&self, provider: PeerId, has: Self::Update) -> PutResult {
+    async fn put(&self, provider: PeerId, mut has: Self::Update) -> PutResult<Self::Update> {
         let span = tracing::info_span!("Peer::LocalStorage::put");
         let _guard = span.enter();
 
         match has.urn.proto {
             uri::Protocol::Git => {
                 let peer_id = has.origin.unwrap_or_else(|| provider);
+                has.origin = Some(peer_id);
+                let urn = Either::Right(Originates {
```

Comment:
Just tested this out with @xla and the seed and it seems like the fetch fails on the seed node.